src/toxcore/onion/onion_client.rs

```diff
@@ -358,12 +377,15 @@ impl OnionClient {
         let mut futures = Vec::with_capacity(payload.nodes.len());
 
         for node in &payload.nodes {
-            // TODO: reduce pings frequency? (see good_to_ping in c-toxcore)
-
             if !nodes_list.can_add(&announce_packet_data.search_pk, &node, /* evict */ true) {
                 continue;
             }
 
+            // To prevent to send redundant ping packet.
+            if self.is_redundant_ping(node.pk, announce_packet_data.search_pk, &state.announce_requests) {
```

Comment:
Why did you add self as the first arg? It was not here before. Try to call clippy  on the project

Reply:
https://github.com/tox-rs/tox/blob/00a77960257b865aa29fc4e662817d9021498a68/src/toxcore/onion/onion_client.rs#L302

Now self is needed.

Reply:
OK